### PR TITLE
Add config

### PR DESCRIPTION
**Pull request by D4nte**
Introduces a configuration file and start using it.
There are still a few things to do but I feel like right now it is a good pit stop.
Done:
- Max sell amounts
- dai contract address
- spread

Yet to be done (in a different PR)
- seed file
- fee reserve

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+use comit::ethereum::ChainId;
+use libp2p::Multiaddr;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use url::Url;
+
+pub use self::{file::File, settings::Settings};
+
+lazy_static::lazy_static! {
+    pub static ref LND_URL: Url = Url::parse("https://localhost:8080").expect("static string to be a valid url");
```
Comment (da-kami) [inline src/config.rs:18]: guess that is a copy left over :)

File: src/config/settings.rs [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                level: Some(level.into()),
+            }),
+            bitcoin: Some(bitcoin.into()),
+            ethereum: Some(ethereum.into()),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct HttpApi {
```
Comment (da-kami) [inline src/config/settings.rs:105]: Don't think we need this. More parts from this file can be removed.

File: src/config/serde/bitcoin_amount.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+use crate::bitcoin::*;
+use serde::{Deserialize, Deserializer, Serializer};
+
+// pub fn deserialize<'de, D>(deserializer: D) -> Result<Amount, D::Error>
```
Comment (da-kami) [inline src/config/serde/bitcoin_amount.rs:4]: ❓ That supposed to be still there?

File: src/main.rs [hunk 4]
```diff
@@ -60,15 +64,25 @@ async fn init_maker(
...
+        .expect("Could not initialize configuration");
+
+    let dai_contract_addr: comit::ethereum::Address = settings.ethereum.dai_contract_address;
 
     // TODO: Proper wallet initialisation from config
-    let bitcoin_wallet =
-        bitcoin_wallet::Wallet::new(unimplemented!(), unimplemented!(), unimplemented!()).unwrap();
-    let ethereum_wallet = ethereum_wallet::Wallet::new(unimplemented!(), unimplemented!()).unwrap();
+    let bitcoin_wallet = bitcoin_wallet::Wallet::new(
+        unimplemented!(),
```
Comment (da-kami) [inline src/main.rs:77]: ❓  Where does the seed come from, do we just use `seed.rs` here?

Comment (D4nte) [reply src/main.rs:77]: Read the description :) The seed is not done.

File: src/config/serde/bitcoin_amount.rs (hunk 3 above)
Comment (D4nte) [reply src/config/serde/bitcoin_amount.rs:4]: oh thanks! Yeah I wanted to try something at some point, I'll try now.

File: src/main.rs (hunk 4 above)
Comment (da-kami) [reply src/main.rs:77]: Sorry, missed that one! :)

**Pull request merged**